devtools/changelog.py

```diff
@@ -0,0 +1,143 @@
+#!/usr/bin/env python3
+from collections import namedtuple
+import re
+import sys
+import shlex
+import subprocess
+import requests
+from mako.template import Template
+import argparse
+from datetime import datetime
+
+# What sections do we support in the changelog:
+sections = [
+    'added',
+    'changed',
+    'deprecated',
+    'fixed',
+    'removed',
+    'security',
+]
+
+repo = 'ElementsProject/lightning'
+
+Entry = namedtuple("Entry", ["commit", "pullreq", "content", "section"])
+
+
+def git(cmd):
+    cmd = shlex.split(cmd)
+    out = subprocess.check_output(['git'] + cmd)
+    return out.decode('UTF-8')
+
+
+def get_commit_range():
+    """Find a commit range that we should collect the CHANGELOG for.
+    """
+    description = git("describe")
+    version = description.split('-')[0]
+    return "{version}..master".format(version=version)
+
+
+def get_log_entries(commitrange):
+    commit = None
+    logs = git("log {commitrange}".format(commitrange=commitrange))
+    entries = []
+
+    for l in logs.split('\n'):
+        m = re.match(r'^commit ([A-Fa-f0-9]{40})$', l)
+        if m:
+            commit = m.group(1)
+
+        m = re.match(
+            r'^\s+Changelog-({}): (.*)$'.format("|".join(sections)), l)
+        if not m:
```

Comment:
Hmm, would be nice to have a warning if I misspell the section name after Changelog-?

Reply:
Would it make more sense to have that check in the bot? At the time we are generating the changelog using this script it's likely too late to actually go and fix it since the commit was merged into master and presumably buried under all the other changes that came afterwards.

Reply:
Ack!